File: stripe/src/main/java/com/stripe/android/PaymentIntentResult.java
package com.stripe.android;

import androidx.annotation.NonNull;

import com.stripe.android.model.PaymentIntent;

public final class PaymentIntentResult extends StripeIntentResult<PaymentIntent> {

    private PaymentIntentResult(@NonNull Builder builder) {
        super(builder.mPaymentIntent, builder.mOutcome);
    }

    static final class Builder implements ObjectBuilder<PaymentIntentResult> {
        private PaymentIntent mPaymentIntent;
        @Outcome private int mOutcome;

        @NonNull
        Builder setPaymentIntent(@NonNull PaymentIntent paymentIntent) {
            mPaymentIntent = paymentIntent;
            return this;
        }

        @NonNull
        Builder setOutcome(@Outcome int outcome) {
            mOutcome = outcome;
            return this;
        }

        @NonNull
        public PaymentIntentResult build() {
            return new PaymentIntentResult(this);
        }
    }
}

File: stripe/src/main/java/com/stripe/android/PaymentIntentResult.kt
package com.stripe.android

import com.stripe.android.model.PaymentIntent
import kotlinx.parcelize.Parcelize

/**
 * A model representing the result of a [PaymentIntent] confirmation via [Stripe.confirmPayment]
 * or handling of next actions via [Stripe.handleNextActionForPayment].
 */
@Parcelize
data class PaymentIntentResult internal constructor(
    override val intent: PaymentIntent,
    @Outcome override val outcomeFromFlow: Int = 0,
    val failureMessage: String? = null
) : StripeIntentResult<PaymentIntent>()
